tests/e2e/Services/Sites/SitesCustomServerTest.php

```diff
@@ -65,6 +65,149 @@ public function testCreateSite(): void
+        $this->assertEquals(true, $secretVariable['body']['secret']);
+
+        // update variable
+        $variable = $this->updateVariable($siteId, $variable['body']['$id'], [
+            'key' => 'siteKey1Updated',
+            'value' => 'siteValue1Updated',
+        ]);
+
+        $this->assertEquals(200, $variable['headers']['status-code']);
+        $this->assertNotEmpty($variable['body']['$id']);
+        $this->assertEquals('siteKey1Updated', $variable['body']['key']);
+        $this->assertEquals('siteValue1Updated', $variable['body']['value']);
+
+        // update variable to secret
+        $variable = $this->updateVariable($siteId, $variable['body']['$id'], [
+            'key' => 'siteKey1Updated',
+            'secret' => true,
+        ]);
+
+        $this->assertEquals(200, $variable['headers']['status-code']);
+        $this->assertNotEmpty($variable['body']['$id']);
+        $this->assertEquals('siteKey1Updated', $variable['body']['key']);
+        $this->assertEquals('', $variable['body']['value']);
+        $this->assertEquals(true, $variable['body']['secret']);
+
+        // update value of secret variable
+        $secretVariable = $this->updateVariable($siteId, $secretVariable['body']['$id'], [
+            'key' => 'siteKey3',
+            'value' => 'siteValue3Updated',
+            'secret' => true
+        ]);
+
+        $this->assertEquals(200, $secretVariable['headers']['status-code']);
+        $this->assertNotEmpty($secretVariable['body']['$id']);
+        $this->assertEquals('siteKey3', $secretVariable['body']['key']);
+        $this->assertEquals('', $secretVariable['body']['value']);
+        $this->assertEquals(true, $secretVariable['body']['secret']);
+
+        // update secret variable to non-secret
+        $temp = $this->updateVariable($siteId, $secretVariable['body']['$id'], [
+            'key' => 'siteKey3',
+            'secret' => false,
+        ]);
+
+        $this->assertEquals(400, $temp['headers']['status-code']);
+
+        // get secret variable
+        $secretVariable = $this->getVariable($siteId, $secretVariable['body']['$id']);
+
+        $this->assertEquals(200, $secretVariable['headers']['status-code']);
+        $this->assertNotEmpty($secretVariable['body']['$id']);
+        $this->assertEquals('siteKey3', $secretVariable['body']['key']);
+        $this->assertEquals('', $secretVariable['body']['value']);
+        $this->assertEquals(true, $secretVariable['body']['secret']);
+
+        // list variables
+        $variables = $this->listVariables($siteId);
+
+        $this->assertEquals(200, $variables['headers']['status-code']);
+        $this->assertCount(3, $variables['body']['variables']);
+
+        // delete variable
+        $this->deleteVariable($siteId, $variable['body']['$id']);
+        $this->deleteVariable($siteId, $variable2['body']['$id']);
+        $this->deleteVariable($siteId, $secretVariable['body']['$id']);
```

Comment:
Lets include assertions

Reply:
I've asserted the count of variables after deleting

Reply:
Let's also have a simple 204 assertion after each deletion.
Having that helps pin-point what exactly goes wrong in test, when it fails in future due to some changes.
We can keep count afterwards too, sounds logical as well